### ISSUE TITLE
Some classmethods defined elsewhere cannot be called normally

### ISSUE THREAD

**Issue by Shunsum** [pyscf/pyscf#2223]
1. `mf = mol.X2C_KS()` does not work with an AttributeError like this
```python
>>> from pyscf import gto
>>> mol = gto.M(atom='Li', basis='sto-3g', spin=1)
>>> mf = mol.X2C_KS()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/share/home/liuxx/miniconda3/lib/python3.12/site-packages/pyscf/gto/mole.py", line 3735, in __getattr__
    raise AttributeError(f'Mole object has no attribute {key}')
AttributeError: Mole object has no attribute X2C_KS. Did you mean: 'X2C_HF'?
```
though the attribute I've found is defined in https://github.com/pyscf/pyscf/blob/7d94887d2ca817287226c51371845037f4469325/pyscf/dft/__init__.py#L120-L128.
It's worth noting that `mol.X2C_KS()` could be normally called without any errors raised if PySCF is manually built from source.

2. Similarly, `mf.gen_response()` in X2C-SCF approach cannot be called either, and it says there's no attribute 'gen_response' found,
```python
>>> from pyscf import gto
>>> mol = gto.M(atom='Li', basis='sto-3g', spin=1)
>>> mf = mol.X2C()
>>> mf.kernel()

WARN: HOMO -0.0463389858396905 == LUMO -0.0463389858347049

converged SCF energy = -7.31595504150337
-7.3159550415033685
>>> mf.gen_response()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'UHF' object has no attribute 'gen_response'
```
while it occurs in https://github.com/pyscf/pyscf/blob/7d94887d2ca817287226c51371845037f4469325/pyscf/x2c/_response_functions.py#L23-L31.

**Comment by sunqm** [comment]
Pyscf employs lazy import for dft and many other modules, similar to the issue reported in #2200  .

For a quickfix, you can execute
```
from pyscf import gto, dft
```
before calling `mol.X2C_KS`.

I'll fix this issue in next release

**Comment by sunqm** [comment]
This bug was fixed by PR #2226
